### ISSUE TITLE
an alternative to compiling from source?

### ISSUE THREAD

**Issue by inieves** [docker-library/php#710]
Hello, I am writing to inquire if there is some way to use the PHP docker images without having to go through all the compile steps?  The primary reason is that I do alot of clean docker image builds, and am spending alot of time waiting for PHP to compile.  This is an especially painful process on single core machines (which are plentiful in the cloud).

Cheers,
Ian

**Comment by tianon** [comment]
You're rebuilding the base image?  Why not pull it from Docker Hub instead?

**Comment by inieves** [comment]
Good question.  I am not pulling the image from Docker because I prefer to use "slim" versions of Debian images (which have a much smaller footprint).  So all the images I build/use are:
FROM debian:stretch-slim

I find that PHP only has official Debian images based on stretch.

Your response made made me think that maybe I should build my base stretch-slim PHP image and then push to docker hub, and then reuse from there...  Would that make sense/work?

**Comment by inieves** [comment]
I might as well add a reason for stretch-slim...
because alpine is philosophically awesome (super small footprint)... but it's too divergent from the Debian ecosystem... Many projects I do are significantly easier in apt-get.  So my compromise is the slim line of Debian images... and I have to be happy with that... even though still, alpine is awesomely lightweight.

**Comment by tianon** [comment]
Sure, that's sane, but also: https://github.com/docker-library/php/blob/e92dfe4847c9fe74ea309e66f9d3ef0217f8525d/7.2/stretch/cli/Dockerfile#L7

(As in, these images are already based on Debian's slim variants.)

**Comment by inieves** [comment]
VERY interesting... OK... i wonder if this changed since I looked last... or if I made an oversight.  I will look at that again.

Id like to mention that I modified the stock PHP Dockerfile:
1)  ha.pool.sks-keyservers.net -> ipv4.pool.sks-keyservers.net (because the former is _very_ flaky and causing failed builds [but for some reason only in my mysql images, not PHP images])

This is not a dealbreaker for me, I can go back to using ha.pool... for PHP docker images...
I will give this a try and see how goes!
